packages/webapp/src/delegates/api/eden-contract.ts

```diff
@@ -65,7 +66,8 @@ export const getMyDelegation = async (
         }
         isHeadChief = member.account === member.representative;
         nextMemberAccount = member.representative;
-    } while (isValidDelegate(nextMemberAccount) && !isHeadChief);
+        currRound -= 1;
```

Comment:
You need to decrement this in the for loop as well to ensure you run out of rounds appropriately. That for loop will add members to the Delegation, which means it should reduce the currRound as well.

Reply:
"as well" or "instead of?"

Reply:
Got it. Thanks!